quil-rs/src/parser/lexer/mod.rs

```diff
@@ -568,7 +489,13 @@ mod tests {
                 Token::Identifier("a-2".to_string()),
                 Token::Operator(Operator::Minus),
                 Token::Variable("var".to_string())
-        ])
+        ]),
+        case("BIT", vec![Token::DataType(DataType::Bit)]),
+        case("BITS", vec![Token::Identifier("BITS".to_string())]),
+        case("NaN", vec![Token::Identifier("NaN".to_string())]),
+        case("nan", vec![Token::Identifier("nan".to_string())]),
+        case("NaNa", vec![Token::Identifier("NaNa".to_string())]),
+        case("nana", vec![Token::Identifier("nana".to_string())]),
```

Comment:
would be worth adding cases for `-Infinity, Infinity, -Inf, Inf` to validate the comment from https://github.com/rigetti/quil-rs/pull/428/files#diff-a5bcc7a39fd5fe75b07e825fc96eeaa32b6cbed091f5936b09d97f9e9112769eR174

Reply:
Done, and they correctly pass – thanks!